include/nbl/builtin/hlsl/shared_memory_accessor.hlsl

```diff
@@ -0,0 +1,155 @@
+        accessor.set(ix + _NBL_HLSL_WORKGROUP_SIZE_, value.y);
+        accessor.set(ix + 2 * _NBL_HLSL_WORKGROUP_SIZE_, value.z);
+        accessor.set(ix + 3 * _NBL_HLSL_WORKGROUP_SIZE_, value.w);
+    }
+
+    void set(const uint ix, const int value) {accessor.set(ix, asuint(value));}
+    void set(const uint ix, const int2 value) {
+        accessor.set(ix, asuint(value.x));
+        accessor.set(ix + _NBL_HLSL_WORKGROUP_SIZE_, asuint(value.y));
+    }
+    void set(const uint ix, const int3 value)  {
+        accessor.set(ix, asuint(value.x));
+        accessor.set(ix + _NBL_HLSL_WORKGROUP_SIZE_, asuint(value.y));
+        accessor.set(ix + 2 * _NBL_HLSL_WORKGROUP_SIZE_, asuint(value.z));
+    }
+    void set(const uint ix, const int4 value) {
+        accessor.set(ix, asuint(value.x));
+        accessor.set(ix + _NBL_HLSL_WORKGROUP_SIZE_, asuint(value.y));
+        accessor.set(ix + 2 * _NBL_HLSL_WORKGROUP_SIZE_, asuint(value.z));
+        accessor.set(ix + 3 * _NBL_HLSL_WORKGROUP_SIZE_, asuint(value.w));
+    }
+
+    void set(const uint ix, const float value) {accessor.set(ix, asuint(value));}
+    void set(const uint ix, const float2 value) {
+        accessor.set(ix, asuint(value.x));
+        accessor.set(ix + _NBL_HLSL_WORKGROUP_SIZE_, asuint(value.y));
+    }
+    void set(const uint ix, const float3 value)  {
+        accessor.set(ix, asuint(value.x));
+        accessor.set(ix + _NBL_HLSL_WORKGROUP_SIZE_, asuint(value.y));
+        accessor.set(ix + 2 * _NBL_HLSL_WORKGROUP_SIZE_, asuint(value.z));
+    }
+    void set(const uint ix, const float4 value) {
+        accessor.set(ix, asuint(value.x));
+        accessor.set(ix + _NBL_HLSL_WORKGROUP_SIZE_, asuint(value.y));
+        accessor.set(ix + 2 * _NBL_HLSL_WORKGROUP_SIZE_, asuint(value.z));
+        accessor.set(ix + 3 * _NBL_HLSL_WORKGROUP_SIZE_, asuint(value.w));
+    }
+    
+    // REVIEW:  Depending on how we handle SHARED_MEM in nbl::hlsl::atomics 
+	//			we should also update this part to handle float and int
+    
+    // uint atomics
+    void atomicAdd(const uint ix, const uint value, out uint orig) {
+	   orig = accessor.atomicAdd(ix, value);
+	}
+	void atomicAnd(const uint ix, const uint value, out uint orig) {
+	   orig = accessor.atomicAnd(ix, value);
+	}
+	void atomicOr(const uint ix, const uint value, out uint orig) {
+	   orig = accessor.atomicOr(ix, value);
+	}
+	void atomicXor(const uint ix, const uint value, out uint orig) {
+	   orig = accessor.atomicXor(ix, value);
+	}
+	void atomicMin(const uint ix, const uint value, out uint orig) {
+	   orig = accessor.atomicMin(ix, value);
+	}
+	void atomicMax(const uint ix, const uint value, out uint orig) {
+	   orig = accessor.atomicMax(ix, value);
+	}
+	void atomicExchange(const uint ix, const uint value, out uint orig) {
+	   orig = accessor.atomicExchange(ix, value);
+	}
+	void atomicCompSwap(const uint ix, const uint value, const uint comp, out uint orig) {
+	   orig = accessor.atomicCompSwap(ix, comp, value);
+	}
```

Comment:
the atomics we should do with SPIR-V intrinsics to use `vk::ext_reference`

but still the accessor must provide all these methods as used here.

Reply:
hmmm maybe try `[[vk::ext_reference]]` chains?

Reply:
tried and opened an issue about it:
https://github.com/microsoft/DirectXShaderCompiler/issues/5675